pydantic_numpy/helper/annotation.py

```diff
@@ -30,6 +38,7 @@ def factory(
         data_type: Optional[SupportedDTypes] = None,
         dimensions: Optional[int] = None,
         strict_data_typing: bool = False,
+        serialize_numpy_array_to_json: Callable[[np.ndarray], Any] = _serialize_numpy_array_to_data_dict,
```

Comment:
Why aren't we using `numpy.typing.ArrayLike`?

I always try to steer away from `Any`, can we use `typing.Iterable`?
```suggestion
        serialize_numpy_array_to_json: Callable[[npt.ArrayLike], Iterable] = _serialize_numpy_array_to_data_dict,
```